project/SparkBuild.scala

```diff
@@ -250,7 +250,10 @@ object SparkBuild extends PomBuild {
         // reduce the cost of migration in subsequent versions.
         "-Wconf:cat=deprecation&msg=it will become a keyword in Scala 3:e",
         // SPARK-46938 to prevent enum scan on pmml-model, under spark-mllib module.
-        "-Wconf:cat=other&site=org.dmg.pmml.*:w"
+        "-Wconf:cat=other&site=org.dmg.pmml.*:w",
+        // Some deprecated calls is a compilation error in Scala 2.13.15.
+        "-Wconf:cat=deprecation&msg=^(?=.*?method|value|object|trait|class|constructor)(?=.*?in)(?=.*?deprecated).+$:w",
```

Comment:
We should make this rule more specific, as its current scope of matching is too broad. Additionally, we should leave a TODO to remind us to fix these deprecated API usages.

Reply:
Okay, let's go through Maven's test first.

Reply:
https://github.com/apache/spark/blob/73d6bd7c35b599690d40efe306eea0774f272ba8/project/SparkBuild.scala#L237

I want to know if this line of configuration has become invalid in Scala 2.13.15？Here we have configured `-Wconf:cat=deprecation:wv`.

Reply:
Let me verify it.

Reply:
- Scala 2.13.14 with `-Wconf:cat=deprecation:wv,any:e`,  printed a `warning` message for the use of deprecated API

```
~/Tools/scala-2.13.14/bin/scala -Wconf:cat=deprecation:wv,any:e
Welcome to Scala 2.13.14 (OpenJDK 64-Bit Server VM, Java 17.0.12).
Type in expressions for evaluation. Or try :help.

scala> class Foo {
     |   def isEmpty(): Boolean = true
     | }
class Foo

scala> val foo = new Foo
val foo: Foo = Foo@706ddbc8

scala> val ret = foo.isEmpty
                     ^
       warning: Auto-application to `()` is deprecated. Supply the empty argument list `()` explicitly to invoke method isEmpty,
       or remove the empty argument list from its definition (Java-defined methods are exempt).
       In Scala 3, an unapplied method like this will be eta-expanded into a function. [quickfixable]
       Applicable -Wconf / @nowarn filters for this warning: msg=<part of the message>, cat=deprecation, site=ret, version=2.13.3
val ret: Boolean = true
```

-  Scala 2.13.15 with `-Wconf:cat=deprecation:wv,any:e`,  printed a `error` message for the use of deprecated API

```
~/Tools/scala-2.13.15/bin/scala -Wconf:cat=deprecation:wv,any:e
Welcome to Scala 2.13.15 (OpenJDK 64-Bit Server VM, Java 17.0.12).
Type in expressions for evaluation. Or try :help.

scala> class Foo {
     |   def isEmpty(): Boolean = true
     | }
class Foo

scala> val foo = new Foo
val foo: Foo = Foo@53f0d09c

scala> val ret = foo.isEmpty
                     ^
       error: Auto-application to `()` is deprecated. Supply the empty argument list `()` explicitly to invoke method isEmpty,
       or remove the empty argument list from its definition (Java-defined methods are exempt).
       In Scala 3, an unapplied method like this will be eta-expanded into a function. [quickfixable]
       Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=deprecation, site=ret, version=2.13.3
```

friendly ping @SethTisue Is this a change that was intentionally introduced in Scala 2.13.15?

Reply:
Thanks @SethTisue ~

@panbingkun I think we should test 

```
<arg>-Wconf:any:e</arg>
<arg>-Wconf:cat=deprecation:wv</arg>
```
for maven 

and 

```
"-Wconf:any:e",
"-Wconf:cat=deprecation:wv",
```

for sbt